samples/azure-quantum/resource-estimation/estimation-profiling.ipynb

```diff
@@ -0,0 +1,825 @@
+    "\n",
+    "    # The runtime profile is the first profile\n",
+    "    runtime_profile = profile['profiles'][0]\n",
+    "\n",
+    "    # Get variables to the samples and weights field of the runtime profile\n",
+    "    samples = runtime_profile['samples']\n",
+    "    weights = runtime_profile['weights']\n",
+    "\n",
+    "    # Sum up all the weights that correspond to samples that contain the operation,\n",
+    "    # i.e., that contain the frame_index\n",
+    "    return sum(weight for (sample, weight) in zip(samples, weights) if frame_index in sample)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now extract the profile for each job result item and use the `rounds_runtime`\n",
+    "function to obtain the runtime for each round, add it to a data frame together\n",
+    "with the total runtime and return a plot."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "entries = []\n",
+    "for idx in range(len(results_all)):\n",
+    "    bitwidth = bitwidths[idx]\n",
+    "    result = results_all[idx]\n",
+    "    profile = result['profile']\n",
+    "\n",
+    "    total_runtime = result['physicalCounts']['runtime']\n",
+    "    entries.append([\n",
+    "        rounds_runtime(profile, \"PRounds\"),\n",
+    "        rounds_runtime(profile, \"GRounds\"),\n",
+    "        rounds_runtime(profile, \"CRounds\"),\n",
+    "        total_runtime\n",
+    "    ])\n",
+    "\n",
+    "df = pandas.DataFrame(data=entries, index=bitwidths, columns=[\"PRounds\", \"GRounds\", \"CRounds\", \"Total\"])\n",
+    "ax = df.plot(logx=True, xticks=bitwidths)\n",
+    "# show all xticks\n",
+    "from matplotlib.text import Text\n",
+    "ax.set_xticklabels([Text(b, 0, b) for b in bitwidths]);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note how the total runtime grows much faster compared to the runtime of the\n",
+    "rounds.  The reason is that we need $O(n)$ AND gates in the preparation part of\n",
+    "`LookAheadAdd` but only $O(\\log(n))$ AND and CCNOT gates in the `ComputeCarries`\n",
+    "operation.\n",
+    "\n",
+    "Further note that logical depth of a the carry-lookahead adder is also\n",
+    "logarithmic in $n$, since on the logical level, all AND and CCNOT gates, in both\n",
+    "the preparation parts and in the rounds can be applied in parallel.  However,\n",
+    "when mapping to surface code operations using Parallel Synthesis Sequential\n",
+    "Pauli Computation (PSSPC), these operations are sequentialized (see Appendix D\n",
+    "in [Assessing requirements to scale to practical quantum\n",
+    "advantage](https://arxiv.org/pdf/2211.07629.pdf))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Next steps"
```

Comment:
Add to summary and introduction the new methods/parameters that can now be used in this notebook for profiling so people can refer back without having to go through the notebook to apply profiling to other notebooks